### ISSUE TITLE
Popup for adding screenshot on the translate page

### ISSUE THREAD

**Issue by nijel** [WeblateOrg/weblate#5583]
**Is your feature request related to a problem? If so, please describe.**

Right now user has to go to the screenshot management to upload screenshot for a string. This is lengthy and requires user to search for the string while adding.

**Describe the solution you'd like**

Popup dialog on the translate page that would allow uploading a screenshot for current string.

**Describe alternatives you've considered**

The current solution works, but is far from being user-friendly.

**Comment by comradekingu** [comment]
How about everyone gets to upload screenshots, and admins are tasked with handling the management UI and actually confirming the screenshots for use?

**Comment by nijel** [comment]
That's another topic, see https://github.com/WeblateOrg/weblate/issues/3472. This is to make upload simple for users who have the permission.

**Comment by comradekingu** [comment]
It is sort of the same topic, because "simple" questions why they shouldn't.

It trades doing the same manual operation many times for avoiding the full UI.
Without access to the full UI, the same picture would be duplicated for many strings.

I would just use the regular search to find more of the strings in each picture and upload many times I think. I like the glossaries are components, everything is the regular way to do things-thinking.

If the checksum is the same, avoiding duplication is possible. Save once, link many strings to it type of deal.

This just means there needs to be a possibility of replacing old images, possibly when uploading, but just a button for each uploaded image keeps it simple.

At that point I am not sure the full UI is going to be used as much.

Problems:
How is the user then aware of all the other active links? If you delete one link, chances are more strings are affected. In the very least it is sleeker (and space-saving) to have other strings in the same UI reflect current state, even if the string in question is correct.

Put in a little text saying "you can also update other strings found in your picture to ensure they reflect the current state of the UI"?

Assuming there that the UI is somewhat deterministic, and that the user isn't an idiot.

Does anyone keep around old screenshots to have some kind of UI progression log? Could still do it, but it would be harder if users got to upload live. If strings and even source strings can be reverted, I don't see how screenshots are more prone to malice. If anything I'd want people to risk their accounts without the danger of it ending up in the software.

Users could be sneaky and change both the source string _and_ the image, but why?

---
TL;DR Uploads on a string-per-string basis with deletion of un-linked images after a while would be nice. That way someone could come in and re-link a "upload to replace" type deal if two screenshots are valid. (String exists two places).

Seems to me it is easy enough to get it good, but very difficult to get it perfect without using a full UI approach. 

Similarly, strings could be considered recent if the screenshot was added for the current iteration of a string, and get a "needs work" checkbox if fuzzified. That way they don't have to be tagged with releases or similar.

**Comment by comradekingu** [comment]
![bilde](https://user-images.githubusercontent.com/13802408/110219655-e70ad700-7ec0-11eb-9600-7f69d72d4b23.png)

I think this would be nice. At the bottom it is out of view if there are many screenshots, which seems like a good idea, but I would want new images to appear at the top, where the + is.

**Comment by nijel** [comment]
We can do more complex things later. This is really about adding a popup form to the translate page which would upload a screenshot, add current string and bring the user to full screenshot editing. This is easy to implement and improves the workflow a lot.

**Comment by comradekingu** [comment]
Yeah the not saving things twice is just finesse for a later day.

So onclick for images in side panel
![bilde](https://user-images.githubusercontent.com/13802408/110219841-d6a72c00-7ec1-11eb-8135-380a9fbcd167.png)

That keeps it manageable for phone-users.

edit "rediger" could be removed or grayed out for regular users.

**Comment by nijel** [comment]
> The full screenshot UI is a bit obscure I think.

I agree, but that's not a topic of this issue. There already is #3838, #3817, #3617, #3472 or #3366.
